lib/internal/modules/esm/worker.js

```diff
@@ -97,6 +97,17 @@ async function customizedModuleWorker(lock, syncCommPort, errorHandler) {
     // so it can detect the exit event.
     const { exit } = process;
     process.exit = function(code) {
+      if (hooks) {
+        for (const registeredPort of allThreadRegisteredHandlerPorts) {
```

Comment:
```suggestion
        for (let i = 0; i < allThreadRegisteredHandlerPorts.length; i++) {
          const registeredPort = allThreadRegisteredHandlerPorts[i];
```
https://github.com/nodejs/node/blob/b0443690f89816ca3b9b8dee7e73f4527b3ff3e0/doc/contributing/primordials.md#L144-L202

Reply:
This is the worker, which I think we said does not need to use primordials?

Reply:
This is a classic for loop, not primordials. I don't think there's any reason not to use it, last we check it's still more performant than `for of` loops – and more reliable in case user hooks have mutated built-ins.

Reply:
sorry, i don't know what happened. my comment was meant for your other suggestion in this file.

Reply:
>  last we check it's still more performant than for of loop

It is? If I remember correctly v8 translates for...of to exactly regular for loops on arrays (unless Array.prototype was mutated or it's a subclass)